### ISSUE TITLE
Stack trace order is backwards

### ISSUE THREAD

**Issue by mindplay-dk** [sebastianbergmann/phpunit#2044]
The order in which stack traces are displayed by php-unit is backwards compared with php's own error and exception stack traces - for example, error and exception stack traces look like this in php 7.0:

```
$ php -r "function foo() { throw new RuntimeException; } function bar() { foo(); } function baz() { bar(); } baz();"
PHP Fatal error:  Uncaught RuntimeException in Command line code:1
Stack trace:
#0 Command line code(1): foo()
#1 Command line code(1): bar()
#2 Command line code(1): baz()
#3 {main}
  thrown in Command line code on line 1

Fatal error: Uncaught RuntimeException in Command line code:1
Stack trace:
#0 Command line code(1): foo()
#1 Command line code(1): bar()
#2 Command line code(1): baz()
#3 {main}
  thrown in Command line code on line 1
```

Having to "switch modes" and figure out the order is very confusing - during a normal day, you see php error or exception stack traces, stack-traces from JavaScript consoles (Chrome, Node, etc.) and maybe stack-traces from other languages (like Java or C#) all of which are backwards compared with php-unit's stack traces.

Given that stack traces generated by php itself (as well as by most other languages) are displayed head-first, I suggest reversing the stack trace print-outs for consistency with, well, everything else.

Note that I'm aware that the stack trace order is consistent with that of xdebug. However, it too is inconsistent with php itself, as is clear from the [second print-out on this page](https://gist.github.com/mindplay-dk/d50205f38f1d0cf3744f), where you actually see two stack-tracks in opposing order on screen at the same time. I have [reported](http://bugs.xdebug.org/view.php?id=1259) that issue against xdebug as well.

While you can argue in favor of one direction or the other, those arguments, in my opinion, are nowhere near as important as mere consistency - the mental switch required to read stack traces in opposing orders costs time. Even if you don't care about being consistent with other languages, it's tough to argue for inconsistency with php itself, and probably with JS, which most PHP developers use too.

**Comment by mindplay-dk** [comment]
@sebastianbergmann any thoughts on this?

@derickr doesn't feel like fixing it, but I'm hoping maybe you feel differently - and that he might change his mind, if you decide to correct php-unit.

This inconsistency is hugely distracting, and it doesn't get better with experience. I wish everyone (php-unit, codeception, xdebug) would just agree on what a stack trace looks like and stick with it. Aligning with php itself would seem like the natural thing to do.

If you don't feel like doing the work either, let me know if you'd accept a PR...

**Comment by stof** [comment]
The order of linked exceptions is also weird: PHPUnit displays the wrapped previous exception before displaying the exception itself

**Comment by sebastianbergmann** [comment]
All of this is very weird. I'd rather completely remove the filtering of exceptions then try to fix and/or change this, though.

**Comment by sebastianbergmann** [comment]
Closed in favor of #2089.
